src/view_wrapper.tsx

```diff
@@ -65,7 +65,18 @@ export default class ViewWrapper<M> extends React.PureComponent<ViewWrapperProps
     this.setState(this.execContext.push(merge(state, pick(keys(state), childProps))));
   }
 
+  public componentWillReceiveProps(nextProps) {
+    const nextChildProps = nextProps.childProps;
+    const { container, childProps } = this.props;
+    if (!equals(nextChildProps, childProps) && container.accepts(Refresh)) {
+      this.execContext.dispatch(new Refresh(omit(['emit'], nextChildProps), { shallow: true }));
+    }
+  }
+
   public componentWillUnmount() {
+    if (this.props.container.accepts(Deactivate)) {
+      this.execContext.dispatch(new Deactivate({ shallow: true }));
```

Comment:
You're passing what should be the second parameter as the first. Maybe it makes sense to pass the current props in here as well though?

Reply:
I could see the case for that.